easybuild/framework/easyconfig/format/format.py

```diff
@@ -370,7 +370,7 @@ def parse(self, configobj):
         for key, value in self.supported.items():
             if key not in self.VERSION_OPERATOR_VALUE_TYPES:
                 raise EasyBuildError('Unsupported key %s in %s section', key, self.SECTION_MARKER_SUPPORTED)
-            self.sections['%s' % key] = value
+            self.sections[key] = value
```

Comment:
This is not fully equivalent, are we sure we don't need `str(key)` here?

Reply:
That question is actually a good reason **for** this change ;-)   
The change is correct: This line is never executed if the key is not also a key in `VERSION_OPERATOR_VALUE_TYPES` which are all strings and cannot reasonably be anything else.   
Alternatively: A non-string key doesn't make sense at this point.

Reply:
OK, thanks for clarifying!